### PR TITLE
Restore vueconfig cleanup with Vue3 migration fixes

### LINKED ISSUE #11744: Restore vue.config.js cleanup as it breaks on extension
# Description

Issue https://github.com/rancher/dashboard/issues/11030 has been reverted and needs to be restored, combined with the fix for https://github.com/rancher/dashboard/issues/9539 as they have been discarded at a certain point while cherry-picking.

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #11744

### Technical notes summary

There's an issue with the extensions caused by this configuration of the aliases and I'm investigating with https://github.com/rancher/dashboard/issues/11706

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/pages/c/_cluster/settings/performance.vue
Comment (rak-phillip) [inline shell/pages/c/_cluster/settings/performance.vue:1]: This changes doesn't appear to be related to the `vue.config` changes. Did we intend to include it in this PR?

**Review by rak-phillip — COMMENTED** [review]
This change looks good and gets us back to the config that we were using in Vue2. Everything that I've tested is looking good for this one. I just have one question about the inclusion of `shell/pages/c/_cluster/settings/performance.vue`

**Review by rak-phillip — APPROVED** [review]

File: shell/pages/c/_cluster/settings/performance.vue
Comment (cnotv) [reply shell/pages/c/_cluster/settings/performance.vue:1]: oops nope

**Review by rak-phillip — DISMISSED** [review]
